lib/ansible/playbook/conditional.py

```diff
@@ -143,6 +144,14 @@ def _check_conditional(self, conditional, templar, all_vars):
         if conditional in all_vars and VALID_VAR_REGEX.match(conditional):
             conditional = all_vars[conditional]
 
+        if self.this is not None:
+            if 'this' in all_vars:
+                display.warning("variable 'this' is a reserved keyword and should not be set")
+            else:
+                # Avoid polluting all_vars
+                all_vars = all_vars.copy()
```

Comment:
this is already a copy ... tempted to just add it in for templating and removing it after.

Reply:
Yup, I used to believe that too, but still if I did not make that copy, on a subsequent call Ansible would believe "this" already was defined and warns about it. So it doesn't seem to be true.

Reply:
Adding and removing is a possibility, yes. But then we need a flag to keep track if we added it or not. (It still has to work for people using **this** as a variable already)

Although at some point I would like to Deprecate the use of the reserved keyword, whatever it is decided to be.

Reply:
which is why we normally use `ansilbe_` to ensure the reserved namespace, there would be 'no deprecation period' ... but i know you loathe the extra typing

Reply:
No, I favor convenience and usability :-)

Reply:
understood, but namespacing is also a usability and convenience issue, it makes it easy for the user to define variables 'safe from conflicts'.

Reply:
Sure. I think we can match both world-views ;-)